eth/api_tracer.go

```diff
@@ -561,9 +562,45 @@ func (api *PrivateDebugAPI) standardTraceBlockToFile(ctx context.Context, block
 
 	// Execute transaction, either tracing all or just the requested one
 	var (
-		signer = types.MakeSigner(api.eth.blockchain.Config(), block.Number())
-		dumps  []string
+		signer      = types.MakeSigner(api.eth.blockchain.Config(), block.Number())
+		dumps       []string
+		chainConfig = api.eth.blockchain.Config()
+		canon       = true
 	)
+	// Check if there are any overrides: the caller may wish to enable a future
+	// fork when executing this block. Note, such overrides are only applicable to the
+	// actual specified block, not any preceding blocks that we have to go through
+	// in order to obtain the state.
+	// Therefore, it's perfectly valid to specify `"futureForkBlock": 0`, to enable `futureFork`
+
+	if config != nil && config.Overrides != nil {
+		// Copy the config, to not screw up the main config
+		chainConfigCopy := &params.ChainConfig{
+			ChainID:             chainConfig.ChainID,
+			HomesteadBlock:      chainConfig.HomesteadBlock,
+			DAOForkBlock:        chainConfig.DAOForkBlock,
+			DAOForkSupport:      chainConfig.DAOForkSupport,
+			EIP150Block:         chainConfig.EIP150Block,
+			EIP150Hash:          chainConfig.EIP150Hash,
+			EIP155Block:         chainConfig.EIP150Block,
+			EIP158Block:         chainConfig.EIP158Block,
+			ByzantiumBlock:      chainConfig.ByzantiumBlock,
+			ConstantinopleBlock: chainConfig.ConstantinopleBlock,
+			PetersburgBlock:     chainConfig.PetersburgBlock,
+			IstanbulBlock:       chainConfig.IstanbulBlock,
+			MuirGlacierBlock:    chainConfig.MuirGlacierBlock,
+			YoloV2Block:         chainConfig.YoloV2Block,
+			EWASMBlock:          chainConfig.EWASMBlock,
+			Ethash:              chainConfig.Ethash,
+			Clique:              chainConfig.Clique,
+		}
```

Comment:
```
chainConfigCopy := new(params.ChainConfig)
*chainConfigCopy = *chainConfig
```

^ future proof, just add a mention that it not a deep copy in the `Clique` struct

Reply:
Or alternatively we need a copy in `params`, otherwise this is 100% going stale.